onnxruntime/core/graph/model.cc

```diff
@@ -359,13 +382,21 @@ static Status SaveModel(Model& model, const T& file_path) {
   int fd;
   Status status = Env::Default().FileOpenWr(file_path, fd);
   ORT_RETURN_IF_ERROR(status);
-  try {
+
+  auto sub_status = Status::OK();
+  ORT_TRY {
     status = Model::Save(model, fd);
-  } catch (const std::exception& ex) {
-    GSL_SUPPRESS(es .84)
-    ORT_IGNORE_RETURN_VALUE(Env::Default().FileClose(fd));
-    return Status(ONNXRUNTIME, FAIL, ex.what());
   }
+  ORT_CATCH(const std::exception& ex) {
+    ORT_HANDLE_EXCEPTION([&]() {
+      GSL_SUPPRESS(es .84)
+      ORT_IGNORE_RETURN_VALUE(Env::Default().FileClose(fd));
+      sub_status = Status(ONNXRUNTIME, FAIL, ex.what());
+    });
```

Comment:
Same question here. Can we just set status?

Reply:
If we do there will be logic change,
The original logic, If we catch exc, then we directly return an error status
If we use status, then before   ORT_RETURN_IF_ERROR(status) we need to run the code inside   if (!status.IsOK()) {} below,
This will be a different logic


---
In reply to: [476241972](https://github.com/microsoft/onnxruntime/pull/4894#discussion_r476241972) [](ancestors = 476241972)